urllib3/connectionpool.py

```diff
@@ -687,12 +687,17 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                 retries = retries.increment(method, url, response=response, _pool=self)
             except MaxRetryError:
                 if retries.raise_on_redirect:
-                    # Release the connection for this response, since we're not
-                    # returning it to be released manually.
+                    # Drain and release the connection for this response, since
+                    # we're not returning it to be released manually.
+                    response.read()
                     response.release_conn()
```

Comment:
Hmm, yeah, I was sorta thinking it would be nice if the response did this for us -- it wouldn't make sense to release a connection back to the pool with data still in the pipe and it seems a bit awkward that a consumer of this lib would have to write the same pair of calls in their code.